src/facade/dragonfly_connection.cc

```diff
@@ -1383,11 +1406,17 @@ void Connection::DispatchFiber(util::FiberSocketBase* peer) {
       RecycleMessage(std::move(msg));
     }
 
-    queue_backpressure_->ec.notify();
+    DCHECK(queue_backpressure_ == &tl_queue_backpressure_);
+    if (stats_->dispatch_queue_bytes < queue_backpressure_->pipeline_buffer_limit / 2) {
+      queue_backpressure_->pipeline_cnd.notify_all();
+    }
+    if (stats_->dispatch_queue_subscriber_bytes < queue_backpressure_->subscriber_thread_limit)
+      queue_backpressure_->ec.notify();
```

Comment:
you said notificaitons are cheap 😆 Moreover, this doesn't reduce their number for the average case. If we want to do so, we can check before whether we were above the limit

Reply:
yes, but I am still trying to understand the threading access pattern around pubsub. `Connection::SendAsync` is always called on a connection thread imho. So why `queue_backpressure_->subscriber_bytes` is atomic? Do we really need thread safety there?

Reply:
Because we check backpressure from a foreign thread, remember we dispatch brief for sending messages, not in a fiber. Should be commented somewhere